### ISSUE TITLE
no-cond-assign incorrectly flagging pattern

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#505]
The `no-cond-assign` rule is incorrect flagging this pattern:

``` js
if ((foo = bar)) {
    // ...
}
```

The double parens are an indicator of the intent to assign (per JSLint) and should not cause a warning.
